src/main/java/net/neoforged/neoforge/data/event/GatherDataEvent.java

```diff
@@ -28,25 +28,27 @@
 import net.neoforged.bus.api.Event;
 import net.neoforged.fml.ModContainer;
 import net.neoforged.fml.event.IModBusEvent;
-import net.neoforged.neoforge.common.data.ExistingFileHelper;
+import net.neoforged.neoforge.common.data.DataResourceManager;
 
 public abstract class GatherDataEvent extends Event implements IModBusEvent {
     private final DataGenerator dataGenerator;
     private final DataGeneratorConfig config;
-    private final ExistingFileHelper existingFileHelper;
     private final ModContainer modContainer;
 
-    public GatherDataEvent(final ModContainer mc, final DataGenerator dataGenerator, final DataGeneratorConfig dataGeneratorConfig, ExistingFileHelper existingFileHelper) {
+    public GatherDataEvent(final ModContainer mc, final DataGenerator dataGenerator, final DataGeneratorConfig dataGeneratorConfig) {
         this.modContainer = mc;
         this.dataGenerator = dataGenerator;
         this.config = dataGeneratorConfig;
-        this.existingFileHelper = existingFileHelper;
     }
 
     public ModContainer getModContainer() {
         return this.modContainer;
     }
 
+    public DataResourceManager getResourceManager() {
```

Comment:
This method should take a `PackType` and return the respective `ResourceManager` directly instead of returning a custom type that basically just duplicates a bunch of methods for very little benefit.